js/ui/status/keyboard.js

```diff
@@ -480,6 +483,53 @@ const InputSourceManager = new Lang.Class({
             this._updateMruSettings();
     },
 
+    get passwordModeEnabled() {
+        return this._passwordModeEnabled;
+    },
+
+    set passwordModeEnabled(enable) {
+        this._passwordModeEnabled = enable;
+        this._switchToLatinLayout();
+    },
+
+    _switchToLatinLayout: function() {
```

Comment:
This function does not actually always switch to latin layout; it will also switch back to the old layout in case password mode is disabled.
I would improve it this way:
```
set passwordModeEnabled(enable) {
  if (enable == this._passwordEnabled)
    return;

  this._syncPasswordMode();
},

_syncPasswordMode: function() {
  // the code that is now in _switchToLatinLayout()
}
```

Reply:
I have renamed the function to _syncPasswordMode. So the early return you have above does not work as I have to clean up things when switch back to the "original" source.